pyccel/ast/utilities.py

```diff
@@ -76,32 +70,44 @@ def builtin_function(expr, args=None):
 
     return None
 
+#==============================================================================
+decorators_mod = Module('decorators',(),
+        funcs = [PyccelFunctionDef(d, PyccelInternalFunction) for d in pyccel_decorators.__all__])
+pyccel_mod = Module('pyccel',(),(),
+        imports = [Import('decorators', decorators_mod)])
 
 # TODO add documentation
-builtin_import_registery = {'numpy': {
-                                      **numpy_functions,
-                                      **numpy_constants,
-                                      'linalg':numpy_linalg_functions,
-                                      'random':numpy_random_functions
-                                      },
-                            'numpy.linalg': numpy_linalg_functions,
-                            'numpy.random': numpy_random_functions,
-                            'scipy.constants': scipy_constants,
-                            'itertools': {'product': Product},
-                            'math': {**math_functions, ** math_constants},
-                            'pyccel.decorators': pyccel_decorators.__all__}
+builtin_import_registery = Module('__main__',
+        (),(),
+        imports = [
+            Import('numpy', AsName(numpy_mod,'numpy')),
+            Import('scipy', AsName(scipy_mod,'scipy')),
+            Import('itertools', AsName(itertools_mod,'itertools')),
+            Import('math', AsName(math_mod,'math')),
+            Import('pyccel', AsName(pyccel_mod,'pyccel'))
+            ])
 if sys.version_info < (3, 10):
     from .builtin_imports import python_builtin_libs
 else:
     python_builtin_libs = set(sys.stdlib_module_names) # pylint: disable=no-member
 
-recognised_libs = python_builtin_libs.union(builtin_import_registery.keys())
+recognised_libs = python_builtin_libs | builtin_import_registery.keys()
 
-#==============================================================================
-def collect_relevant_imports(func_dictionary, targets):
-    if len(targets) == 0:
-        return func_dictionary
+def recognised_source(source_name):
+    """ Determine whether the imported source is recognised by pyccel.
+    If it is not recognised then it should be imported and translated
+    """
+    source = str(source_name).split('.')
+    if source[0] in python_builtin_libs and source[0] not in builtin_import_registery.keys():
+        return True
+    else:
+        return source_name in builtin_import_registery
 
+#==============================================================================
+def collect_relevant_imports(func_module, targets):
+    """
+    Extract all objects necessary to create imports from a module given a list of targets
+    """
```

Comment:
Could you please expand this docstring and describe that the output of this function is a list of tuple pairs `(local_name, imported_object)`?

Also, is `func_module` of type `Module`? If yes, we could change its name to just `module`...

Reply:
Done cf9f6e76